src/main/java/io/seqera/tower/agent/Agent.java

```diff
@@ -182,17 +188,47 @@ private void sendPeriodicHeartbeat() {
     }
 
     private void sendInfoMessage() throws IOException {
-        String userName = new UnixSystem().getUsername();
-        String workDir = this.workDir.toAbsolutePath().normalize().toString();
-        String agentVersion = getVersion();
-
         agentClient.send(new InfoMessage(
-                userName,
-                workDir,
-                agentVersion
+                validatedUserName,
+                validatedWorkDir,
+                getVersion()
         ));
     }
 
+    private void validateParameters() throws IOException {
+        // Fetch username
+        validatedUserName = System.getProperty("user.name");
+        if (validatedUserName == null || validatedUserName.isEmpty() || validatedUserName.isBlank()) {
+            logger.error("Impossible to detect current Unix username.");
+            System.exit(-1);
+        }
+
+        // Set default workDir
+        if (workDir == null) {
+            String defaultPath = System.getProperty("user.home") + "/work";
+            try {
+                workDir = Paths.get(defaultPath);
+            } catch (InvalidPathException e) {
+                logger.error("Impossible to define a default work directory. Please provide one using '--work-dir'.");
+                System.exit(-1);
```

Comment:
-1 is weird. It should be 1, no ?

Reply:
I've always seen -1 (=255) on error but let's use 1 as we are doing on Tower CLI.